React/Modules/RCTTiming.m

```diff
@@ -233,16 +233,18 @@ - (void)scheduleSleepTimer:(NSDate *)sleepTarget
     _sleepTimer = [[NSTimer alloc] initWithFireDate:sleepTarget
                                            interval:0
                                              target:[_RCTTimingProxy proxyWithTarget:self]
-                                           selector:@selector(timerDidFire)
+                                           selector:@selector(sleepTimerDidFire)
                                            userInfo:nil
                                             repeats:NO];
-    [[NSRunLoop currentRunLoop] addTimer:_sleepTimer forMode:NSDefaultRunLoopMode];
+    dispatch_async(dispatch_get_main_queue(), ^{
```

Comment:
To clarify, are you saying you'd prefer to use `[NSRunLoop mainRunLoop]` rather than the dispatch methods?